app-backend/api/src/main/scala/user/Routes.scala

```diff
@@ -135,7 +135,7 @@ trait UserRoutes
   def getUserByEncodedAuthId(authIdEncoded: String): Route = authenticate {
     user =>
       rejectEmptyResponse {
-        val authId = URLDecoder.decode(authIdEncoded, "US_ASCII")
+        val authId = URLDecoder.decode(authIdEncoded, "US-ASCII")
```

Comment:
Agreed that it is unclear how this was working. There must be an alias setup somewhere along the path. I'm more worried that this should actually be UTF-8 vs. ASCII.

Reply:
I took a closer look, and it seems that `URLDecoder` is not directly invoking `Charset.forName`. Down the chain, it actually creates a new instance of `String`:

```java
sb.append(new String(bytes, 0, pos, enc));
```

Which invokes `StringDecoder.decode`:

```java
        this.value = StringCoding.decode(charsetName, bytes, offset, length);
```

It looks like there's some broken logic in `StringCoding.decode` where it will fallback to a default value if the encoding passed is invalid, rather than throw an `UnsupportedEncodingException` as it should. I believe that `US_ASCII` is invalid in JDK8, and it has been falling back to `ISO-8859-1`.

See:

```java
    static char[] decode(String charsetName, byte[] ba, int off, int len)
        throws UnsupportedEncodingException
    {
        StringDecoder sd = deref(decoder);
        String csn = (charsetName == null) ? "ISO-8859-1" : charsetName;
        if ((sd == null) || !(csn.equals(sd.requestedCharsetName())
                              || csn.equals(sd.charsetName()))) {
            sd = null;
            try {
                Charset cs = lookupCharset(csn);
                if (cs != null)
                    sd = new StringDecoder(cs, csn);
            } catch (IllegalCharsetNameException x) {}
            if (sd == null)
                throw new UnsupportedEncodingException(csn);
            set(decoder, sd);
        }
        return sd.decode(ba, off, len);
    }
```

I think that switching to `UTF-8` makes sense, and would resolve any weird encoding errors we may run into in the future:

```suggestion
        val authId = URLDecoder.decode(authIdEncoded, "UTF-8")
```